test/api/projects_api_test.rb

```diff
@@ -11,11 +11,33 @@ def app
   end
 
   def test_can_get_projects
-    user = FactoryBot.create(:user, :student, enrol_in: 0)
-    get with_auth_token('/api/projects', user)
+    # Create new student user
+    user = FactoryBot.create(:user, :admin)
+
+    # Create a dummy project
+    new_project = FactoryBot.create(:project)
+
+    # Perform Get
+    get with_auth_token "/api/projects/#{new_project.id}", new_project.user
+
+    # Check if the call success
     assert_equal 200, last_response.status
   end
 
+  def test_cannot_get_projects
+    # A user with student role
+    user = FactoryBot.create(:user, :student)
+
+    # Create a dummy project
+    project = FactoryBot.create(:project)
+  
+    # Perform Get, but cannot find Project with id=project.id
+    get with_auth_token "/api/projects/#{project.id}", user
```

Comment:
This GET is not through because you are assigning a student from a different project to perform the request, meaning it is not true a user with student role cannot get project's details, it has to be the right user of the project. You might want to delete this test, work on the other test to check a student can get project's details.